sami2py/_core_class.py

```diff
@@ -174,7 +159,27 @@ def _load_model(self):
         nf = 98
         nz = 101
         ni = 7
-
+        
+        def return_fourier(x, coeffs):
+            """
+            Returns a Fourier series up to NumF coefficients
+            """
+            def cos_a(x, n):
+                """simple cosine"""
+                return np.cos(n * np.pi * x / 12.0)
+        
+            def sin_a(x, n):
+                """simple sine"""
+                return np.sin(n * np.pi * x / 12.0)
+        
+            NumF = int((len(coeffs) - 1)/2)
+        
+            y = coeffs[0]
+            for i in range(1, NumF+1):
+                y = y + coeffs[2*i-1]*cos_a(x, i) + coeffs[2*i]*sin_a(x, i)
```

Comment:
The coeffs matrix at this point is 10x2.  You'd have to you a reshape command to get coeffs in the right shape.
https://numpy.org/doc/stable/reference/generated/numpy.reshape.html

Or potentially just rewrite this as `coeffs[i, 0]` and `coeffs[i, 1]` if we keep the coefficient matrix as is.